homeassistant/components/homekit/type_thermostats.py

```diff
@@ -236,27 +250,45 @@ def _set_chars(self, char_values):
         state = self.hass.states.get(self.entity_id)
         features = state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
 
-        hvac_mode = self.hass.states.get(self.entity_id).state
+        hvac_mode = state.state
         homekit_hvac_mode = HC_HASS_TO_HOMEKIT[hvac_mode]
 
         if CHAR_TARGET_HEATING_COOLING in char_values:
             # Homekit will reset the mode when VIEWING the temp
             # Ignore it if its the same mode
             if char_values[CHAR_TARGET_HEATING_COOLING] != homekit_hvac_mode:
                 target_hc = char_values[CHAR_TARGET_HEATING_COOLING]
-                if target_hc in self.hc_homekit_to_hass:
-                    service = SERVICE_SET_HVAC_MODE_THERMOSTAT
-                    hass_value = self.hc_homekit_to_hass[target_hc]
-                    params = {ATTR_HVAC_MODE: hass_value}
-                    events.append(
-                        f"{CHAR_TARGET_HEATING_COOLING} to {char_values[CHAR_TARGET_HEATING_COOLING]}"
-                    )
-                else:
-                    _LOGGER.warning(
-                        "The entity: %s does not have a %s mode",
-                        self.entity_id,
-                        target_hc,
-                    )
+                if target_hc not in self.hc_homekit_to_hass:
+                    # If the target heating cooling state we want does not
+                    # exist on the device, we have to sort it out
+                    # based on the the current and target temperature since
+                    # siri will always send HC_HEAT_COOL_AUTO in this case
+                    # and hope for the best.
+                    hc_target_temp = char_values.get(CHAR_TARGET_TEMPERATURE)
+                    hc_current_temp = _get_current_temperature(state, self._unit)
+                    hc_fallback_order = HC_HEAT_COOL_PREFER_HEAT
+                    if (
+                        hc_target_temp is not None
+                        and hc_current_temp is not None
+                        and hc_target_temp < hc_current_temp
+                    ):
+                        hc_fallback_order = HC_HEAT_COOL_PREFER_COOL
+                    for hc_fallback in hc_fallback_order:
+                        if hc_fallback in self.hc_homekit_to_hass:
+                            _LOGGER.debug(
+                                "Siri requested target mode: %s and the device does not support, falling back to %s",
+                                target_hc,
+                                hc_fallback,
+                            )
+                            target_hc = hc_fallback
+                            break
+
+                service = SERVICE_SET_HVAC_MODE_THERMOSTAT
+                hass_value = self.hc_homekit_to_hass[target_hc]
```

Comment:
Can you still get here with a target_hc that is not in hc_homekit_to_hass? I was imagining a case where a loop through hc_fallback_order does not have any matches of hc_fallback in hc_homekit_to_has and target_hc remains unchanged.

Reply:
I see a few of the tests exercise some ValueError cases, so maybe that is intended?

Reply:
> Can you still get here with a target_hc that is not in hc_homekit_to_hass? I was imagining a case where a loop through hc_fallback_order does not have any matches of hc_fallback in hc_homekit_to_has and target_hc remains unchanged.

If the device doesn't support any hvac modes it's not conforming to the climate entity model so I think it should be ok.